validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/KeystoreLocker.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.client.loader;
+
+import static java.nio.file.StandardOpenOption.CREATE_NEW;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.teku.util.config.InvalidConfigurationException;
+
+public class KeystoreLocker {
+
+  private static final Logger LOG = LogManager.getLogger();
+  private final byte[] processPID = longPidToNativeByteArray(ProcessHandle.current().pid());
+
+  public void lockKeystoreFile(Path keystoreFile) {
+    deleteIfStaleLockfileExists(keystoreFile);
+    try {
+      final Path keystoreLockFile =
+          Files.write(Path.of(keystoreFile.toString() + ".lock"), processPID, CREATE_NEW);
```

Comment:
We should have a single place where we create the path to the lock file instead of duplicating it here and in `deleteIfStaleLockfileExists`.

Reply:
Makes sense.